PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/appservice/AppServiceComboBox.java

```diff
@@ -12,72 +12,45 @@
 import com.microsoft.azure.toolkit.intellij.common.AzureComboBox;
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
 import com.microsoft.azuretools.azurecommons.helpers.Nullable;
-import com.microsoft.tooling.msservices.components.DefaultLoader;
+import lombok.Getter;
+import lombok.Setter;
 import org.apache.commons.lang3.StringUtils;
-import org.apache.commons.lang3.exception.ExceptionUtils;
 import rx.Subscription;
 
 import javax.swing.*;
-import java.io.InterruptedIOException;
+import java.util.List;
 import java.util.Objects;
 
-import static com.microsoft.intellij.util.RxJavaUtils.unsubscribeSubscription;
-
 public abstract class AppServiceComboBox<T extends AppServiceComboBoxModel> extends AzureComboBox<T> {
 
     protected Project project;
     protected Subscription subscription;
 
+    @Getter
+    @Setter
+    private T configModel;
+
     public AppServiceComboBox(final Project project) {
         super(false);
         this.project = project;
         this.setRenderer(new AppCombineBoxRender(this));
     }
 
-    // todo: optimize refreshing logic
-    public synchronized void refreshItemsWithDefaultValue(@NotNull T defaultValue) {
-        unsubscribeSubscription(subscription);
-        this.setLoading(true);
-        this.removeAllItems();
-        this.addItem(defaultValue);
-        subscription = this.loadItemsAsync()
-            .subscribe(items -> DefaultLoader.getIdeHelper().invokeLater(() -> {
-                synchronized (AppServiceComboBox.this) {
-                    AppServiceComboBox.this.removeAllItems();
-                    items.forEach(this::addItem);
-                    this.resetDefaultValue(defaultValue);
-                    this.setLoading(false);
-                }
-            }), this::handleLoadingError);
-    }
-
-    private void resetDefaultValue(@NotNull T defaultValue) {
-        final AppServiceComboBoxModel model = getItems()
-            .stream()
-            .filter(item -> AppServiceComboBoxModel.isSameApp(defaultValue, item) && item != defaultValue)
-            .findFirst().orElse(null);
-        if (model != null) {
-            this.setSelectedItem(model);
-            this.removeItem(defaultValue);
-        } else if (defaultValue.isNewCreateResource()) {
-            return;
-        } else {
-            this.setSelectedItem(null);
-            this.removeItem(defaultValue);
-        }
-    }
-
+    @NotNull
     @Override
-    protected void handleLoadingError(final Throwable e) {
-        final Throwable rootCause = ExceptionUtils.getRootCause(e);
-        if (rootCause instanceof InterruptedIOException || rootCause instanceof InterruptedException) {
-            // Swallow interrupted exception caused by unsubscribe
-            return;
+    protected List<? extends T> loadItems() throws Exception {
+        final List<T> items = loadAppServiceModels();
+        if (super.getItemReference() != null && configModel != null && configModel.isNewCreateResource()) {
```

Comment:
remove super.

Reply:
this is a general problem, we shall not resolve the combo box problem in AppServiceComboBox but AzureComboBox

Reply:
`getItemReference` removed, only process it in `AppServiceComboBox`